services/oauth/v2/types.go

```diff
@@ -0,0 +1,108 @@
+package v2
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	rudderSync "github.com/rudderlabs/rudder-go-kit/sync"
+	"github.com/rudderlabs/rudder-server/services/controlplane/identity"
+)
+
+type (
+	RudderFlow string
+	AuthType   string
+	ContextKey string
+)
+
+type expirationDate struct {
+	ExpirationDate string `json:"expirationDate"`
+}
+
+// AccountSecret is the access token returned by the oauth server
+type AccountSecret struct {
+	ExpirationDate string          `json:"expirationDate"`
+	Secret         json.RawMessage `json:"secret"`
+}
+type OAuthHandler struct {
+	TokenProvider
+	Logger                    logger.Logger
+	RudderFlowType            RudderFlow
+	CpConn                    ControlPlaneConnectorI
+	AuthStatusUpdateActiveMap map[string]bool // Used to check if a authStatusInactive request for a destination is already InProgress
+	Cache                     Cache
+	CacheMutex                *rudderSync.PartitionRWLocker
+	ExpirationTimeDiff        time.Duration
+}
+type CacheKey struct {
+	WorkspaceID string
+	AccountID   string
+}
+
+type TokenProvider interface {
+	Identity() identity.Identifier
+}
+
+type ControlPlaneRequestT struct {
```

Comment:
No need to suffix it with `T`
```suggestion
type ControlPlaneRequest struct {
```